core/src/components/segment-button/segment-button.tsx

```diff
@@ -87,15 +87,7 @@ export class SegmentButton implements ComponentInterface, ButtonInterface {
   }
 
   private get tabIndex() {
-    if (this.disabled) { return -1; }
-
-    const hasTabIndex = this.el.hasAttribute('tabindex');
-
-    if (hasTabIndex) {
-      return this.el.getAttribute('tabindex');
-    }
-
-    return 0;
+    return this.checked && !this.disabled ? 0 : -1;
```

Comment:
Have we verified that this does not regress https://github.com/ionic-team/ionic-framework/pull/23145?

Reply:
Yeah, I found no regression!

Additionally, this follows the `tabIndex` pattern shown here: https://www.w3.org/TR/wai-aria-practices-1.1/examples/tabs/tabs-1/tabs.html#sc1_label